lib/gitlab/request.rb

```diff
@@ -62,28 +67,25 @@ def validate(response)
     # Sets a base_uri and default_params for requests.
     # @raise [Error::MissingCredentials] if endpoint not set.
     def request_defaults(sudo = nil)
-      self.class.default_params sudo: sudo
       raise Error::MissingCredentials, 'Please set an endpoint to API' unless @endpoint
 
+      self.class.default_params sudo: sudo
```

Comment:
I moved this below the `raise` while I was debugging some cross-test pollution; see https://github.com/NARKOZ/gitlab/pull/513/files#diff-a36c72b98dfda3970f82ee56454344e3R12-R13.

I'm not sure why `self.class.default_params(sudo: sudo) unless sudo.nil?` didn't work, but oh well. 🤷‍♂